src/devices/Ina226/Ina226.cs

```diff
@@ -0,0 +1,489 @@
+    {
+        private I2cDevice _i2cDevice;
+        private bool _disposeI2cDevice = false;
+
+        // cache for commonly used configuration values
+        private ushort _calibrationValue;
+        private Ina226SamplesAveraged _samplesAveraged;
+        private Ina226BusVoltageConvTime _busConvTime;
+        private Ina226ShuntConvTime _shuntConvTime;
+        private int _busConvTimeInt;
+        private int _shuntConvTimeInt;
+        private int _samplesAveragedInt;
+        private double _currentLsb;
+
+        /// <summary>
+        /// Construct an Ina226 device using an I2cDevice
+        /// </summary>
+        /// <remarks>
+        /// This binding does not dispose the passed in I2cDevice.
+        /// </remarks>
+        /// <param name="i2cDevice">The I2cDevice initialized to communicate with the INA226</param>
+        public Ina226(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            _currentLsb = 1F;
+        }
+
+        /// <summary>
+        /// Construct an INA226 device useing I2cConnectionSettings passed in.
+        /// </summary>
+        /// <remarks>
+        /// This binding creates an I2cDevice for communication with the INA226. This I2cDevice is disposed when the INA226 is disposed.
+        /// </remarks>
+        /// <param name="settings">The I2cConnectionSettings object initialized with the appropriate settings to communicate with the INA226</param>
+        public Ina226(I2cConnectionSettings settings)
+            : this(I2cDevice.Create(settings)) => _disposeI2cDevice = true;
+
+        /// <summary>
+        /// Dispose of instance
+        /// </summary>
+        public void Dispose()
+        {
+            if (_disposeI2cDevice)
+            {
+                _i2cDevice?.Dispose();
+                _i2cDevice = null!;
+            }
+        }
+
+        /// <summary>
+        /// Reset the INA226 to default values.
+        /// </summary>
+        [Command]
+        public void Reset()
+        {
+            // Reset the device by sending a value to the configuartion register with the reset bit set.
+            WriteRegister(Ina226Register.Configuration, (ushort)Ina226ConfigurationFlags.Reset);
+
+            _currentLsb = 1F;
+            _calibrationValue = 0;
+
+            // cache the values for the Bus, Shunt, and Samples Averaged settings so that they can be used later without reading them from the INA226 device.
+            _samplesAveraged = SamplesAveraged;
+            _samplesAveragedInt = int.Parse(_samplesAveraged.ToString().Replace("Quantity_", string.Empty));
+
+            _busConvTime = BusConvTime;
+            _busConvTimeInt = int.Parse(_busConvTime.ToString().Replace("Time", string.Empty).Replace("us", string.Empty));
+
+            _shuntConvTime = ShuntConvTime;
+            _shuntConvTimeInt = int.Parse(_shuntConvTime.ToString().Replace("Time", string.Empty).Replace("us", string.Empty));
```

Comment:
This parsing here is not nice. It's slow and very error prone. If somebody decides to change enum names, this breaks undetected. And the logic is even duplicated in the properties below. Use a lookup table or a dictionary instead.

Reply:
replaced this with switch methods and caching to remove the parsing and .ToString calls. also removed from the properties below it as well.